http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala

```diff
@@ -754,6 +754,7 @@ object Boot extends IOApp {
         .withConnectTimeout(30 seconds)
         .withRequestTimeout(60 seconds)
         .withMaxTotalConnections(100)
+        .withMaxIdleDuration(30 seconds)
```

Comment:
Doc for this field:
```
  * @param maxIdleDuration maximum time a connection can be idle and still
  * be borrowed.  Helps deal with connections that are closed while
  * idling in the pool for an extended period.
```

The previous (and default) setting was `Infinite`.

Our theory is that http4s blaze-client maintains a connection pool with connections that never expire. However spring-based services (like WSM) have an idle timeout of 1 minute. So Leo was getting connections from the pool that had already been terminated, leading to `HTTP connection closed`. 

Try setting the idle duration to a smaller value to see if it has a positive impact. Traffic should be low enough that reestablishing connections shouldn't be that much overhead. Note this affects _all_ Leo clients, not just WSM. Planning to do some BEE-load-testing of this.

Reply:
Testing in a BEE, I managed to get `HTTP connection closed` (from Sam, not WSM) even with this change. The Sam call was a GET though, so it retried and succeeded.

Thinking of making it retry all "connection closed" errors, regardless of method (GET/POST).